Neos.Media/Classes/Domain/Repository/ImageVariantRepository.php

```diff
@@ -0,0 +1,66 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Media\Domain\Repository;
+
+/*
+ * This file is part of the Neos.Media package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Media\Domain\Model\ImageVariant;
+
+/**
+ * A repository for ImageVariants
+ *
+ * @Flow\Scope("singleton")
+ */
+class ImageVariantRepository extends AssetRepository
+{
+    /**
+     * Returns array of ImageVariants with outdated presets
+     *
+     * @param String[] $configuredPresets
+     * @return ImageVariant[]
+     */
+    public function findAllWithOutdatedPresets(array $configuredPresets, bool $isCliPresetFilter, bool $filterCroppedVariants, int $limit = null): array
+    {
+        $configuredIdentifiers = array_keys($configuredPresets);
+        $queryBuilder = $this->entityManager->createQueryBuilder()
+            ->select('iv')
+            ->from(ImageVariant::class, 'iv')
+            ->setMaxResults($limit);
+
+        if (!$isCliPresetFilter) {
```

Comment:
This is unclear to me, maybe I need a coffee to understand the use of this, it seems to contradict the method name to not filter?